analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -125,6 +133,27 @@ private Message buildMessage(MessageBuilder builder) {
     return message;
   }
 
+  private void addExternalContext(String contextKey, Object contextValue) {
+    this.externalContext = new HashMap<>();
```

Comment:
this is a private method - and I can't see aanything calling it? Also it replaces the external context every time so it would only ever have 1 key?